source/server/admin/admin.h

```diff
@@ -244,6 +246,37 @@ class AdminImpl : public Admin,
     TimeSource& time_source_;
   };
 
+  /**
+   * Implementation of OverloadManager that is never overloaded. Using this instead of the real
+   * OverloadManager keeps the admin interface accessible even when the proxy is overloaded.
+   */
+  struct NullOverloadManager : public OverloadManager {
+    struct NullThreadLocalOverloadState : public ThreadLocalOverloadState {
+      const OverloadActionState& getState(const std::string&) override { return inactive_; }
+
+      const OverloadActionState inactive_ = OverloadActionState::Inactive;
+    };
+
+    NullOverloadManager(ThreadLocal::SlotAllocator& slot_allocator)
+        : tls_(slot_allocator.allocateSlot()) {}
+
+    void start() override {
```

Comment:
is start() called?

Reply:
Nope, moved the TLS initialization to the constructor, which seems to work fine.

Reply:
I think you opted to use start() in the end.

Reply:
What was the reasoning to keep this in start vs the ctor?

Reply:
The constructor gets invoked with AdminImpl's constructor, which is invoked in server.cc before the TLS state is initialized. Putting this in start lets us defer it.